.github/workflows/build-test.yml

```diff
@@ -0,0 +1,42 @@
+name: Build and test
+
+on:
+  push:
+    branches:
+      - '**'
+    tags-ignore:
+      - 'v*'
+
+  pull_request:
+    branches:
+      - '**'
+    tags-ignore:
+      - 'v*'
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Setup Java
+        uses: actions/setup-java@v4
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+
+      - name: Build with Gradle
+        run: ./gradlew build
+
+      - name: Run Test
+        run: ./gradlew test
+
+      - name: Upload build artifacts
```

Comment:
Why upload this?

Reply:
@stephane-segning 
To upload the artifact generated during a workflow run sir.